src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -219,6 +234,18 @@ private static boolean hasAtLeastOneSiblingWithSameTokenType(DetailAST ast) {
     }
 
     /**
+     * Checks if the given {@code DetailAST} node is matching line and column number.
+     * @param ast {@code DetailAST} root ast
+     * @return true if the given {@code DetailAST} node is matching line and column number
+     */
+    private boolean isMatchingByLineAndColumn(DetailAST ast) {
+        return ast.getLineNo() == lineNumber
+                && 1 + CommonUtils.lengthExpandedTabs(fileText.get(lineNumber - 1),
+            ast.getColumnNo(), tabWidth) == columnNumber
+                && ast.getType() != TokenTypes.IDENT;
```

Comment:
0) 
> isMatchingByLineAndColumn

What about IDENT , as you do  `ast.getType() ! = tokenTypes.IDENT` .... please do not cheat in names.

1) as all parts of expression is combined by `&&`  - please put the most simple parts to be executed first.

2) `1+ ...... ` that is very complicated expression and it damage ability to read it clearly. WE need to make it singleline.
please make it like:
```
ast.getType() ! = tokenTypes.IDENT
&& ast.getLineNo == lineNumber
&& expandedTabColumn(ast) == columnNumber
```

Reply:
done